plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/controller/CodeTransformChatController.kt

```diff
@@ -665,9 +666,21 @@ class CodeTransformChatController(
         codeTransformChatHelper.addNewMessage(buildStartNewTransformFollowup())
     }
 
+    private suspend fun handleCodeTransformJobFailed(failureReason: String) {
+        codeTransformChatHelper.updateLastPendingMessage(buildTransformFailedChatContent(failureReason))
+        codeTransformChatHelper.addNewMessage(buildStartNewTransformFollowup())
+    }
+
+    private suspend fun handleCodeTransformJobFailedPreBuild(result: CodeModernizerJobCompletedResult.JobFailedInitialBuild) =
+        codeTransformChatHelper.addNewMessage(
+            buildTransformResultChatContent(result)
+        )
+
     private suspend fun handleCodeTransformResult(result: CodeModernizerJobCompletedResult) {
         when (result) {
             is CodeModernizerJobCompletedResult.Stopped, CodeModernizerJobCompletedResult.JobAbortedBeforeStarting -> handleCodeTransformStoppedByUser()
+            is CodeModernizerJobCompletedResult.JobFailed -> handleCodeTransformJobFailed(result.failureReason)
+            is CodeModernizerJobCompletedResult.JobFailedInitialBuild -> handleCodeTransformJobFailedPreBuild(result)
```

Comment:
These make sure that when the job fails / when the server-side build fails, the last chat item is updated to say so